[GTK][WPE] Enable -Wunsafe-buffer-usage for the WebCore target

aperezdc opened the pull request:
#### 4993aa174c49f7b5bfc0bc91069440cca262be16
<pre>
[GTK][WPE] Enable -Wunsafe-buffer-usage for the WebCore target
<a href="https://bugs.webkit.org/show_bug.cgi?id=282597">https://bugs.webkit.org/show_bug.cgi?id=282597</a>

Reviewed by Chris Dumez.

Enable additional bounds safety checks from -Wunsafe-buffer-usage for
the WebCore target for the GTK and WPE ports. There is a fair bit of
WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} suppressions which need to be
added, to be addressed later on. For the issues inside inline functions
from Skia headers, use WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_{BEGIN,END}
around their inclusion.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/encryptedmedia/InitDataRegistry.cpp:
(WebCore::isPlayReadySanitizedInitializationData):
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransformerOpenSSL.cpp:
(WebCore::crypt):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::initializePipeline):
(WebCore::transceiverStatesFromWebRTCBin):
(WebCore::GStreamerMediaEndpoint::pickAvailablePayloadType):
(WebCore::GStreamerMediaEndpoint::transceiverBackendFromSender):
(WebCore::GStreamerMediaEndpoint::collectTransceivers):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerPeerConnectionBackend.cpp:
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
(WebCore::parseIceCandidateSDP):
* Source/WebCore/Modules/webaudio/MediaStreamAudioSourceGStreamer.cpp:
(WebCore::copyBusData):
* Source/WebCore/accessibility/atspi/AccessibilityAtspi.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESCBCGCrypt.cpp:
(WebCore::gcryptEncrypt):
(WebCore::gcryptDecrypt):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESCTRGCrypt.cpp:
(WebCore::gcryptAESCTR):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESGCMGCrypt.cpp:
(WebCore::gcryptDecrypt):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
(WebCore::gcryptVerify):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
(WebCore::verifyEd25519):
* Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
(WebCore::exponentVectorToUInt32):
* Source/WebCore/crypto/gcrypt/GCryptUtilities.cpp:
(WebCore::mpiZeroPrefixedData):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleTouchEvent):
* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::cpuPeriod):
(WebCore::threadCPUUsage):
* Source/WebCore/platform/SharedBufferChunkReader.cpp:
* Source/WebCore/platform/audio/SincResampler.cpp:
(WebCore::SincResampler::convolve): Deleted.
* Source/WebCore/platform/audio/VectorMath.cpp:
* Source/WebCore/platform/audio/glib/MediaSessionGLib.cpp:
(WebCore::MediaSessionGLib::ensureMprisSessionRegistered):
* Source/WebCore/platform/audio/gstreamer/AudioDestinationGStreamer.cpp:
(WebCore::AudioDestinationGStreamer::AudioDestinationGStreamer):
* Source/WebCore/platform/audio/gstreamer/AudioEncoderGStreamer.cpp:
(WebCore::GStreamerInternalAudioEncoder::initialize):
* Source/WebCore/platform/audio/gstreamer/AudioFileReaderGStreamer.cpp:
(WebCore::copyGstreamerBuffersToAudioChannel):
* Source/WebCore/platform/audio/gstreamer/FFTFrameGStreamer.cpp:
(WebCore::FFTFrame::doFFT):
(WebCore::FFTFrame::doInverseFFT):
* Source/WebCore/platform/audio/gstreamer/PlatformRawAudioDataGStreamer.cpp:
(WebCore::PlatformRawAudioData::copyTo):
* Source/WebCore/platform/audio/gstreamer/WebKitWebAudioSourceGStreamer.cpp:
(webKitWebAudioSrcAllocateBuffer):
* Source/WebCore/platform/graphics/PixelBufferConversion.cpp:
* Source/WebCore/platform/graphics/WOFFFileFormat.cpp:
(WebCore::readUInt32):
(WebCore::readUInt16):
(WebCore::convertWOFFToSfnt):
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createContextForEGLVersion):
(WebCore::GLContext::isExtensionSupported):
* Source/WebCore/platform/graphics/filters/skia/FEColorMatrixSkiaApplier.cpp:
(WebCore::FEColorMatrixSkiaApplier::apply const):
* Source/WebCore/platform/graphics/gbm/DRMDeviceManager.cpp:
(WebCore::drmForeachDevice):
(WebCore::DRMDeviceManager::initializeMainDevice):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::ensureGStreamerInitialized):
(WebCore::registerWebKitGStreamerElements):
* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::fillVideoRtpCapabilities):
* Source/WebCore/platform/graphics/gstreamer/GstAllocatorFastMalloc.cpp:
(gstMemoryFastMallocNew):
(gstAllocatorFastMallocMemCopy):
(gstAllocatorFastMallocMemIsSpan):
* Source/WebCore/platform/graphics/gstreamer/ImageGStreamerSkia.cpp:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
* Source/WebCore/platform/graphics/gstreamer/TextCombinerPadGStreamer.cpp:
(webkit_text_combiner_pad_class_init):
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrame::createNV12):
(WebCore::VideoFrame::createI420):
(WebCore::VideoFrame::createI420A):
(WebCore::VideoFrame::copyTo):
* Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp:
* Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp:
(webKitWebSrcGetProtocols):
* Source/WebCore/platform/graphics/gstreamer/eme/GStreamerEMEUtilities.cpp:
(WebCore::markupStartElement):
(WebCore::markupEndElement):
* Source/WebCore/platform/graphics/gstreamer/eme/GStreamerEMEUtilities.h:
(WebCore::ProtectionSystemEvents::ProtectionSystemEvents):
* Source/WebCore/platform/graphics/gstreamer/eme/WebKitThunderDecryptorGStreamer.cpp:
* Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaDescription.cpp:
(WebCore::GStreamerMediaDescription::extractCodecName const):
* Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.cpp:
* Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.h:
* Source/WebCore/platform/graphics/opentype/OpenTypeVerticalData.cpp:
* Source/WebCore/platform/graphics/skia/ColorSpaceSkia.h:
* Source/WebCore/platform/graphics/skia/ComplexTextControllerSkia.cpp:
(WebCore::ComplexTextController::ComplexTextRun::ComplexTextRun):
* Source/WebCore/platform/graphics/skia/FloatRoundedRectSkia.cpp:
(WebCore::FloatRoundedRect::operator SkRRect const):
* Source/WebCore/platform/graphics/skia/FontCascadeSkia.cpp:
* Source/WebCore/platform/graphics/skia/GradientSkia.cpp:
* Source/WebCore/platform/graphics/skia/GraphicsContextGLSkia.cpp:
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.h:
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp:
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaSurfaceBackend.h:
* Source/WebCore/platform/graphics/skia/ImageBufferUtilitiesSkia.cpp:
* Source/WebCore/platform/graphics/skia/PathSkia.cpp:
(WebCore::PathSkia::applyElements const):
* Source/WebCore/platform/graphics/skia/PathSkia.h:
* Source/WebCore/platform/graphics/skia/PatternSkia.cpp:
* Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp:
* Source/WebCore/platform/graphics/skia/SkiaHarfBuzzFont.cpp:
(WebCore::SkiaHarfBuzzFont::glyphWidths):
* Source/WebCore/platform/graphics/skia/SkiaHarfBuzzFont.h:
* Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp:
(WebCore::BitmapTexture::updateContents):
* Source/WebCore/platform/graphics/texmap/ClipStack.cpp:
(WebCore::ClipStack::addRoundedRect):
* Source/WebCore/platform/graphics/texmap/TextureMapper.cpp:
(WebCore::TextureMapper::beginPainting):
(WebCore::TextureMapper::endPainting):
(WebCore::computeGaussianKernel):
(WebCore::TextureMapper::bindDefaultSurface):
(WebCore::TextureMapper::updateProjectionMatrix):
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::transformedBoundingBox):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayerBufferNativeImage.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayerBufferVideo.cpp:
(WebCore::CoordinatedPlatformLayerBufferVideo::createBufferFromDMABufMemory):
(WebCore::CoordinatedPlatformLayerBufferVideo::createBufferFromGLMemory):
(WebCore::CoordinatedPlatformLayerBufferVideo::paintToTextureMapper):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedTileBuffer.cpp:
* Source/WebCore/platform/gstreamer/GStreamerCodecUtilities.cpp:
(WebCore::GStreamerCodecUtilities::parseHEVCProfile):
* Source/WebCore/platform/gstreamer/GStreamerQuirkBcmNexus.cpp:
(WebCore::GStreamerQuirkBcmNexus::isHardwareAccelerated):
* Source/WebCore/platform/gstreamer/GStreamerQuirkBroadcom.cpp:
(WebCore::GStreamerQuirkBroadcom::configureElement):
(WebCore::GStreamerQuirkBroadcom::isHardwareAccelerated):
* Source/WebCore/platform/gstreamer/GStreamerQuirkRealtek.cpp:
(WebCore::GStreamerQuirkRealtek::isHardwareAccelerated):
* Source/WebCore/platform/gstreamer/GStreamerQuirkRialto.cpp:
* Source/WebCore/platform/gstreamer/GStreamerQuirkWesteros.cpp:
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
(webkit_video_encoder_class_init):
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.h:
(WebKitVideoEncoderBitRateAllocation::setBitRate):
(WebKitVideoEncoderBitRateAllocation::getBitRate const):
* Source/WebCore/platform/gstreamer/WebKitFliteSourceGStreamer.cpp:
(webkitFliteSrcCreate):
* Source/WebCore/platform/gtk/DragImageGtk.cpp:
* Source/WebCore/platform/image-decoders/ScalableImageDecoder.cpp:
* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
(WebCore::BMPImageReader::readInfoHeader):
(WebCore::BMPImageReader::processRLEData):
* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.h:
(WebCore::BMPImageReader::getComponent const):
(WebCore::BMPImageReader::getAlpha const):
* Source/WebCore/platform/image-decoders/gif/GIFImageDecoder.cpp:
(WebCore::GIFImageDecoder::clearFrameBufferCache):
(WebCore::GIFImageDecoder::haveDecodedRow):
* Source/WebCore/platform/image-decoders/gif/GIFImageReader.cpp:
(GIFImageReader::parse):
* Source/WebCore/platform/image-decoders/ico/ICOImageDecoder.cpp:
(WebCore::ICOImageDecoder::setData):
(WebCore::ICOImageDecoder::processDirectoryEntries):
* Source/WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.cpp:
(WebCore::JPEGImageReader::skipBytes):
(WebCore::setPixel):
(WebCore::JPEGImageDecoder::outputScanlines):
* Source/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp:
* Source/WebCore/platform/image-decoders/skia/ImageBackingStoreSkia.cpp:
* Source/WebCore/platform/image-decoders/webp/WEBPImageDecoder.cpp:
(WebCore::WEBPImageDecoder::applyPostProcessing):
* Source/WebCore/platform/libwpe/PlatformPasteboardLibWPE.cpp:
(WebCore::PlatformPasteboard::getTypes const):
(WebCore::PlatformPasteboard::write):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp:
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeAudioSourceGStreamer.cpp:
(WebCore::MockRealtimeAudioSourceGStreamer::reconfigure):
* Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp:
(WebCore::CertificateInfo::summary const):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::SoupNetworkSession::clearOldSoupCache):
* Source/WebCore/platform/soup/PublicSuffixStoreSoup.cpp:
(WebCore::PublicSuffixStore::platformTopPrivatelyControlledDomain const):
* Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp:
(WebCore::TextCheckerEnchant::getGuessesForWord):
* Source/WebCore/platform/text/hyphen/HyphenationLibHyphen.cpp:
* Source/WebCore/platform/xdg/MIMETypeRegistryXdg.cpp:
(WebCore::MIMETypeRegistry::preferredExtensionForMIMEType):
(WebCore::MIMETypeRegistry::extensionsForMIMEType):

Canonical link: <a href="https://commits.webkit.org/286804@main">https://commits.webkit.org/286804@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3dfa81fed98011a99b49d31cd6ff29c60293b2c1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77131 "1 style error") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/56166 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/30046 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/81689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/28413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/79248 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/65314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/4462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/81689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/28413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80198 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/65314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/66211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/81689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/65314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/23709 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/26736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/65314 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/24039 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/83116 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/4511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/4462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/83116 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/4667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/66184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/83116 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/10048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11940 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/4457 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/7273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/4477 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/7912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/6236 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
```diff
@@ -105,8 +105,10 @@ static ExceptionOr<bool> verifyEd25519(const Vector<uint8_t>& key, size_t keyLen
 
     // Construct the sig-val s-expression, extracting the r and s components from the signature vector.
     PAL::GCrypt::Handle<gcry_sexp_t> signatureSexp;
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
```
@geoffreygaren adopted a specific format for pragmas that are port-specific:
https://commits.webkit.org/286279@main

It would be nice to follow it in this patch. Basically:
```
WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GTK/WPE port
```

cdumez commented on Source/WebCore/platform/audio/SincResampler.cpp:
```diff
@@ -293,6 +293,7 @@ void SincResampler::process(std::span<float> destination, size_t framesToProcess
     }
 }
 
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
```
This incorrectly covers the `#if USE(ACCELERATE)` which doesn't warn and is used by Apple ports. We should not silence warnings in valid code.

This comment likely applies elsewhere.

aperezdc replied:
Right, I will double-check when rebasing to make sure there aren't cases like this left.

aperezdc committed:
[GTK][WPE] Enable -Wunsafe-buffer-usage for the WebCore target
https://bugs.webkit.org/show_bug.cgi?id=282597

Reviewed by Chris Dumez.

Enable additional bounds safety checks from -Wunsafe-buffer-usage for
the WebCore target for the GTK and WPE ports. There is a fair bit of
WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} suppressions which need to be
added, to be addressed later on. For the issues inside inline functions
from Skia headers, use WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_{BEGIN,END}
around their inclusion.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/encryptedmedia/InitDataRegistry.cpp:
(WebCore::isPlayReadySanitizedInitializationData):
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransformerOpenSSL.cpp:
(WebCore::crypt):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::initializePipeline):
(WebCore::transceiverStatesFromWebRTCBin):
(WebCore::GStreamerMediaEndpoint::pickAvailablePayloadType):
(WebCore::GStreamerMediaEndpoint::transceiverBackendFromSender):
(WebCore::GStreamerMediaEndpoint::collectTransceivers):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerPeerConnectionBackend.cpp:
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
(WebCore::parseIceCandidateSDP):
* Source/WebCore/Modules/webaudio/MediaStreamAudioSourceGStreamer.cpp:
(WebCore::copyBusData):
* Source/WebCore/accessibility/atspi/AccessibilityAtspi.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESCBCGCrypt.cpp:
(WebCore::gcryptEncrypt):
(WebCore::gcryptDecrypt):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESCTRGCrypt.cpp:
(WebCore::gcryptAESCTR):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAESGCMGCrypt.cpp:
(WebCore::gcryptDecrypt):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
(WebCore::gcryptVerify):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
(WebCore::verifyEd25519):
* Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
(WebCore::exponentVectorToUInt32):
* Source/WebCore/crypto/gcrypt/GCryptUtilities.cpp:
(WebCore::mpiZeroPrefixedData):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleTouchEvent):
* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::cpuPeriod):
(WebCore::threadCPUUsage):
* Source/WebCore/platform/SharedBufferChunkReader.cpp:
* Source/WebCore/platform/audio/SincResampler.cpp:
(WebCore::SincResampler::convolve): Deleted.
* Source/WebCore/platform/audio/VectorMath.cpp:
* Source/WebCore/platform/audio/glib/MediaSessionGLib.cpp:
(WebCore::MediaSessionGLib::ensureMprisSessionRegistered):
* Source/WebCore/platform/audio/gstreamer/AudioDestinationGStreamer.cpp:
(WebCore::AudioDestinationGStreamer::AudioDestinationGStreamer):
* Source/WebCore/platform/audio/gstreamer/AudioEncoderGStreamer.cpp:
(WebCore::GStreamerInternalAudioEncoder::initialize):
* Source/WebCore/platform/audio/gstreamer/AudioFileReaderGStreamer.cpp:
(WebCore::copyGstreamerBuffersToAudioChannel):
* Source/WebCore/platform/audio/gstreamer/FFTFrameGStreamer.cpp:
(WebCore::FFTFrame::doFFT):
(WebCore::FFTFrame::doInverseFFT):
* Source/WebCore/platform/audio/gstreamer/PlatformRawAudioDataGStreamer.cpp:
(WebCore::PlatformRawAudioData::copyTo):
* Source/WebCore/platform/audio/gstreamer/WebKitWebAudioSourceGStreamer.cpp:
(webKitWebAudioSrcAllocateBuffer):
* Source/WebCore/platform/graphics/PixelBufferConversion.cpp:
* Source/WebCore/platform/graphics/WOFFFileFormat.cpp:
(WebCore::readUInt32):
(WebCore::readUInt16):
(WebCore::convertWOFFToSfnt):
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createContextForEGLVersion):
(WebCore::GLContext::isExtensionSupported):
* Source/WebCore/platform/graphics/filters/skia/FEColorMatrixSkiaApplier.cpp:
(WebCore::FEColorMatrixSkiaApplier::apply const):
* Source/WebCore/platform/graphics/gbm/DRMDeviceManager.cpp:
(WebCore::drmForeachDevice):
(WebCore::DRMDeviceManager::initializeMainDevice):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::ensureGStreamerInitialized):
(WebCore::registerWebKitGStreamerElements):
* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::fillVideoRtpCapabilities):
* Source/WebCore/platform/graphics/gstreamer/GstAllocatorFastMalloc.cpp:
(gstMemoryFastMallocNew):
(gstAllocatorFastMallocMemCopy):
(gstAllocatorFastMallocMemIsSpan):
* Source/WebCore/platform/graphics/gstreamer/ImageGStreamerSkia.cpp:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
* Source/WebCore/platform/graphics/gstreamer/TextCombinerPadGStreamer.cpp:
(webkit_text_combiner_pad_class_init):
* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrame::createNV12):
(WebCore::VideoFrame::createI420):
(WebCore::VideoFrame::createI420A):
(WebCore::VideoFrame::copyTo):
* Source/WebCore/platform/graphics/gstreamer/VideoSinkGStreamer.cpp:
* Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp:
(webKitWebSrcGetProtocols):
* Source/WebCore/platform/graphics/gstreamer/eme/GStreamerEMEUtilities.cpp:
(WebCore::markupStartElement):
(WebCore::markupEndElement):
* Source/WebCore/platform/graphics/gstreamer/eme/GStreamerEMEUtilities.h:
(WebCore::ProtectionSystemEvents::ProtectionSystemEvents):
* Source/WebCore/platform/graphics/gstreamer/eme/WebKitThunderDecryptorGStreamer.cpp:
* Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaDescription.cpp:
(WebCore::GStreamerMediaDescription::extractCodecName const):
* Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.cpp:
* Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.h:
* Source/WebCore/platform/graphics/opentype/OpenTypeVerticalData.cpp:
* Source/WebCore/platform/graphics/skia/ColorSpaceSkia.h:
* Source/WebCore/platform/graphics/skia/ComplexTextControllerSkia.cpp:
(WebCore::ComplexTextController::ComplexTextRun::ComplexTextRun):
* Source/WebCore/platform/graphics/skia/FloatRoundedRectSkia.cpp:
(WebCore::FloatRoundedRect::operator SkRRect const):
* Source/WebCore/platform/graphics/skia/FontCascadeSkia.cpp:
* Source/WebCore/platform/graphics/skia/GradientSkia.cpp:
* Source/WebCore/platform/graphics/skia/GraphicsContextGLSkia.cpp:
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.h:
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp:
* Source/WebCore/platform/graphics/skia/ImageBufferSkiaSurfaceBackend.h:
* Source/WebCore/platform/graphics/skia/ImageBufferUtilitiesSkia.cpp:
* Source/WebCore/platform/graphics/skia/PathSkia.cpp:
(WebCore::PathSkia::applyElements const):
* Source/WebCore/platform/graphics/skia/PathSkia.h:
* Source/WebCore/platform/graphics/skia/PatternSkia.cpp:
* Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp:
* Source/WebCore/platform/graphics/skia/SkiaHarfBuzzFont.cpp:
(WebCore::SkiaHarfBuzzFont::glyphWidths):
* Source/WebCore/platform/graphics/skia/SkiaHarfBuzzFont.h:
* Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp:
(WebCore::BitmapTexture::updateContents):
* Source/WebCore/platform/graphics/texmap/ClipStack.cpp:
(WebCore::ClipStack::addRoundedRect):
* Source/WebCore/platform/graphics/texmap/TextureMapper.cpp:
(WebCore::TextureMapper::beginPainting):
(WebCore::TextureMapper::endPainting):
(WebCore::computeGaussianKernel):
(WebCore::TextureMapper::bindDefaultSurface):
(WebCore::TextureMapper::updateProjectionMatrix):
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::transformedBoundingBox):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayerBufferNativeImage.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayerBufferVideo.cpp:
(WebCore::CoordinatedPlatformLayerBufferVideo::createBufferFromDMABufMemory):
(WebCore::CoordinatedPlatformLayerBufferVideo::createBufferFromGLMemory):
(WebCore::CoordinatedPlatformLayerBufferVideo::paintToTextureMapper):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedTileBuffer.cpp:
* Source/WebCore/platform/gstreamer/GStreamerCodecUtilities.cpp:
(WebCore::GStreamerCodecUtilities::parseHEVCProfile):
* Source/WebCore/platform/gstreamer/GStreamerQuirkBcmNexus.cpp:
(WebCore::GStreamerQuirkBcmNexus::isHardwareAccelerated):
* Source/WebCore/platform/gstreamer/GStreamerQuirkBroadcom.cpp:
(WebCore::GStreamerQuirkBroadcom::configureElement):
(WebCore::GStreamerQuirkBroadcom::isHardwareAccelerated):
* Source/WebCore/platform/gstreamer/GStreamerQuirkRealtek.cpp:
(WebCore::GStreamerQuirkRealtek::isHardwareAccelerated):
* Source/WebCore/platform/gstreamer/GStreamerQuirkRialto.cpp:
* Source/WebCore/platform/gstreamer/GStreamerQuirkWesteros.cpp:
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.cpp:
(webkit_video_encoder_class_init):
* Source/WebCore/platform/gstreamer/VideoEncoderPrivateGStreamer.h:
(WebKitVideoEncoderBitRateAllocation::setBitRate):
(WebKitVideoEncoderBitRateAllocation::getBitRate const):
* Source/WebCore/platform/gstreamer/WebKitFliteSourceGStreamer.cpp:
(webkitFliteSrcCreate):
* Source/WebCore/platform/gtk/DragImageGtk.cpp:
* Source/WebCore/platform/image-decoders/ScalableImageDecoder.cpp:
* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
(WebCore::BMPImageReader::readInfoHeader):
(WebCore::BMPImageReader::processRLEData):
* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.h:
(WebCore::BMPImageReader::getComponent const):
(WebCore::BMPImageReader::getAlpha const):
* Source/WebCore/platform/image-decoders/gif/GIFImageDecoder.cpp:
(WebCore::GIFImageDecoder::clearFrameBufferCache):
(WebCore::GIFImageDecoder::haveDecodedRow):
* Source/WebCore/platform/image-decoders/gif/GIFImageReader.cpp:
(GIFImageReader::parse):
* Source/WebCore/platform/image-decoders/ico/ICOImageDecoder.cpp:
(WebCore::ICOImageDecoder::setData):
(WebCore::ICOImageDecoder::processDirectoryEntries):
* Source/WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.cpp:
(WebCore::JPEGImageReader::skipBytes):
(WebCore::setPixel):
(WebCore::JPEGImageDecoder::outputScanlines):
* Source/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp:
* Source/WebCore/platform/image-decoders/skia/ImageBackingStoreSkia.cpp:
* Source/WebCore/platform/image-decoders/webp/WEBPImageDecoder.cpp:
(WebCore::WEBPImageDecoder::applyPostProcessing):
* Source/WebCore/platform/libwpe/PlatformPasteboardLibWPE.cpp:
(WebCore::PlatformPasteboard::getTypes const):
(WebCore::PlatformPasteboard::write):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp:
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeAudioSourceGStreamer.cpp:
(WebCore::MockRealtimeAudioSourceGStreamer::reconfigure):
* Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp:
(WebCore::CertificateInfo::summary const):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::SoupNetworkSession::clearOldSoupCache):
* Source/WebCore/platform/soup/PublicSuffixStoreSoup.cpp:
(WebCore::PublicSuffixStore::platformTopPrivatelyControlledDomain const):
* Source/WebCore/platform/text/enchant/TextCheckerEnchant.cpp:
(WebCore::TextCheckerEnchant::getGuessesForWord):
* Source/WebCore/platform/text/hyphen/HyphenationLibHyphen.cpp:
* Source/WebCore/platform/xdg/MIMETypeRegistryXdg.cpp:
(WebCore::MIMETypeRegistry::preferredExtensionForMIMEType):
(WebCore::MIMETypeRegistry::extensionsForMIMEType):

Canonical link: https://commits.webkit.org/286804@main
